scripts/install/auto-install.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+# auto-install.sh installs git if it isn't installed, then clones POCS and
+# installs its dependencies.
+#
+# To fetch this from github and execute it immediately, run one of these two
+# commands in a bash shell, the first if you have wget installed, the second
+# if you have curl installed:
+#
+# 1a) wget -O - https://raw.githubusercontent.com/panoptes/POCS/develop/scripts/install/auto-install.sh | bash
+# 1b) bash <(wget -O - https://raw.githubusercontent.com/panoptes/POCS/develop/scripts/install/auto-install.sh)
+#
+# 2) bash <(curl -s https://raw.githubusercontent.com/panoptes/POCS/develop/scripts/install/auto-install.sh)
+#
+# To determine whether you have these commands install, type these commands:
+#
+#   which wget
+#   which curl
+
+################################################################################
+# Env vars used for debugging of this script; these allow you to point to your
+# fork of POCS on github, so that you can download your fork instead of the
+# primary repo. There is no support for doing the same with PAWS.
+
+[[ -z "${POCS_GITHUB_USER}" ]] && export POCS_GITHUB_USER="panoptes"
+[[ -z "${POCS_GIT_URL}" ]] && export POCS_GIT_URL="https://github.com/${POCS_GITHUB_USER}/POCS.git"
+[[ -z "${POCS_BRANCH}" ]] && export POCS_BRANCH="develop"
+
+################################################################################
+# Functions COPIED from install-helper-functions.sh
+# Print the disk location of the first arg.
+function safe_which() {
+  type -p "${1}" || /bin/true
+}
+
+# Print a separator bar of # characters.
+function echo_bar() {
+  local terminal_width
+  if [ -n "${TERM}" ] && [ -t 0 ] ; then
+    if [[ -n "$(which resize)" ]] ; then
+      terminal_width="$(resize 2>/dev/null | grep COLUMNS= | cut -d= -f2)"
+    elif [[ -n "$(which stty)" ]] ; then
+      terminal_width="$(stty size 2>/dev/null | cut '-d ' -f2)"
+    fi
+  fi
+  printf "%${terminal_width:-80}s\n" | tr ' ' '#'
+}
+################################################################################
+
+function do_sudo() {
+  if [ "$(id -u -n)" == "root" ] ; then
+    echo "Running ${*}"
+    "$@"
+  else
+    echo "Running sudo ${*}; you may be prompted for your password."
+    (set -x ; sudo "$@")
+  fi
+}
+
+function clone_or_update() {
+  local -r REPO_DIR="${1}"
+  local -r REPO_URL="${2}"
+  local -r BRANCH="${3}"
+
+  echo_bar
+
+  if [ ! -d "${REPO_DIR}/.git" ]
+  then
+    if [ -d "${REPO_DIR}" ]
+    then
+      echo 2> "
+The directory (${REPO_DIR}) already exists, but doesn't appear
+to be a valid git repository. Please remove it (or move it out
+of the way) and re-run this script.
+  "
+      exit 1
+    fi
+    echo "
+Cloning ${REPO_URL} into ${REPO_DIR}
+"
+    (set -x ; git clone "${REPO_URL}" "${REPO_DIR}")
+    cd "${REPO_DIR}"
+    git checkout "${BRANCH}"
+  else
+    echo "
+Pulling the latest software into the worktree at ${REPO_DIR}
```

Comment:
```suggestion
Pulling the latest software into the worktree at ${REPO_DIR}.
```

You do realize you've made me anal about periods in comments now, right? :wink:

Reply:
Excellent! Done in my local repo, will upload shortly.